crates/rolldown_sourcemap/src/lib.rs

```diff
@@ -14,9 +14,9 @@ use rolldown_utils::rustc_hash::FxHashMapExt;
 
 // <https://github.com/rollup/rollup/blob/master/src/utils/collapseSourcemaps.ts>
 #[allow(clippy::cast_possible_truncation)]
-pub fn collapse_sourcemaps(mut sourcemap_chain: Vec<&SourceMap>) -> SourceMap {
+pub fn collapse_sourcemaps(sourcemap_chain: &[&SourceMap]) -> SourceMap {
   debug_assert!(sourcemap_chain.len() > 1);
-  let last_map = sourcemap_chain.pop().expect("sourcemap_chain should not be empty");
+  let last_map = sourcemap_chain.last().expect("sourcemap_chain should not be empty");
```

Comment:
Does this cause a logic change? The previous code won't touch this map again in iterations.

Reply:
Fixed